### ISSUE TITLE
Wazuh 3.8.0 cannot match System.EventID correctly

### ISSUE THREAD

**Issue by jjf012** [wazuh/wazuh#2587]
|Wazuh version|Install type|Install method|Platform|
|---|---|---|---|
| 3.8.0-1 | Agent | Packages | windows2008 R2 |

I want to use sysmon to monitor mimikatz activity, using the following rules and found that there is no correct match.

- [sysmon config](https://github.com/ion-storm/sysmon-config)

- **agent ossec.conf**
```
  <localfile>
	<location>Microsoft-Windows-Sysmon/Operational</location>
	<log_format>eventchannel</log_format>
  </localfile>
```
- **manager local_rule.xml**
```
<group name="sysmon,">
  <rule id="20018" level="0" overwrite="yes">
    <if_sid>20000</if_sid>
    <description>Windows Sysmon event.</description>
    <options>no_full_log</options>
    <field name="EventChannel.System.Channel">Microsoft-Windows-Sysmon/Operational</field>
    <field name="EventChannel.System.EventID">^1$|^2$|^3$|^4$|^5$|^6$|^7$|
                |^8$|^9$|^10$|^11$|^12$|^13$|^14$|^15$|</field>
  </rule>
    <rule id="20185" level="8" overwrite="yes">
        <if_sid>20018</if_sid>
        <field name="EventChannel.System.EventID">^10$</field>
        <description>Sysmon - Event 10: ProcessAccess $(EventChannel.EventData.Description)</description>
        <options>no_full_log</options>
        <group>sysmon_event_10,</group>
    </rule>
    <rule id="20192" level="8" overwrite="yes">
        <if_sid>20018</if_sid>
        <field name="EventChannel.System.EventID">^8$</field>
        <description>Sysmon - Event 8: CreateRemoteThread $(EventChannel.EventData.Description)</description>
        <options>no_full_log</options>
        <group>sysmon_event8,</group>
    </rule>
    <rule id="20194" level="8" overwrite="yes">
        <if_sid>20018</if_sid>
        <field name="EventChannel.System.EventID">^1$</field>
        <description>Sysmon - Event 1: Process creation $(EventChannel.EventData.Description)</description>
        <options>no_full_log</options>
        <group>sysmon_event1,</group> 
   </rule>
</group>
```
I want to use sysmon to monitor mimikatz activity, using the following rules and found that there is no correct match.
When I execute `sekurlsa::msv` in mimikatz, the sysmon event id is **10**, but the matching rule id in wazuh is 20185 instead of 20194, and there is no match to `EventChannel.System.EventID 8`

**Comment by jjf012** [comment]
`EventChannel.System.ProviderName` is `Microsoft-Windows-Sysmon`
`EventChannel.System.Channel` is `Microsoft-Windows-Sysmon/Operational`

**Comment by cristgl** [comment]
Hello @jjf012, 

First of all, Wazuh v3.8.0 was released with a memory leak that will end up with your Windows agent, we have fixed it and some other Sysmon issues for Wazuh v3.9.0, that will be released in the next weeks.

Regarding your problem, could you paste a log from the event you want to be matched? You can find it  at the Sysmon section of the Eventviewer (Applications and Services Logs/Microsoft/Windows/Sysmon/Operational) once it has been generated.

Best regards and sorry for the inconveniences,
Cristina

**Comment by jjf012** [comment]
@cristgl 
I reversed the rules 20185 and 20194 and found interesting things. 

![image](https://user-images.githubusercontent.com/13266057/53138528-8fdb8780-35c1-11e9-8a5e-3ad95f7cce3f.png)

Sysmon event ID 8, can't be seen in wazuh, even though I wrote the rules.

The original log information is as follows.
* sysmon event id 1
```
Process Create:
RuleName: 
UtcTime: 2019-02-18 03:27:13.988
ProcessGuid: {89ed7566-2611-5c6a-0000-00107a7f3607}
ProcessId: 1976
Image: E:\mimikatz_trunk\x64\mimikatz.exe
FileVersion: 2.1.1.0
Description: mimikatz for Windows
Product: mimikatz
Company: gentilkiwi (Benjamin DELPY)
CommandLine: "E:\mimikatz_trunk\x64\mimikatz.exe" 
CurrentDirectory: E:\mimikatz_trunk\x64\
User: WIN-KPO4XAVNB1D\Administrator
LogonGuid: {89ed7566-8fc8-5c62-0000-0020d39e0a00}
LogonId: 0xa9ed3
TerminalSessionId: 2
IntegrityLevel: High
Hashes: MD5=50300DE5E4786530EA603224CCBCBB02,SHA256=23A243A1CE474C4DA90B1003FFCBAF9A3FF25E0787844BFE74C21671FDD8B269,IMPHASH=66EE036DF5FC1004D9ED5E9A94A1086A
ParentProcessGuid: {89ed7566-8fca-5c62-0000-001083e60a00}
ParentProcessId: 2172
ParentImage: C:\Windows\explorer.exe
ParentCommandLine: C:\Windows\Explorer.EXE
```
* sysmon event id 8
```
CreateRemoteThread detected:
RuleName: 
UtcTime: 2019-02-18 03:27:37.971
SourceProcessGuid: {89ed7566-2611-5c6a-0000-00107a7f3607}
SourceProcessId: 1976
SourceImage: E:\mimikatz_trunk\x64\mimikatz.exe
TargetProcessGuid: {89ed7566-8d3a-5c62-0000-00104fb10000}
TargetProcessId: 568
TargetImage: C:\Windows\System32\lsass.exe
NewThreadId: 2328
StartAddress: 0x0000000000100000
StartModule: 
StartFunction: 
```

**Comment by cristgl** [comment]
Hello @jjf012,

The Sysmon event with ID 8 is not matching due to a bug written at the rules that was fixed for the next Wazuh versions already released. This is the failing rule:

```
<rule id="20018" level="0">
    <if_sid>20000</if_sid>
    <description>Windows Sysmon event.</description>
    <options>no_full_log</options>
    <field name="EventChannel.System.Channel">Microsoft-Windows-Sysmon/Operational</field>
    <field name="EventChannel.System.EventID">^1$|^2$|^3$|^4$|^5$|^6$|^7$|
                ^8$|^9$|^10$|^11$|^12$|^13$|^14$|^15$|</field>
</rule>
```
The correct form to filter by these IDs would be the next one:
```
    <field name="EventChannel.System.EventID">^1$|^2$|^3$|^4$|^5$|^6$|^7$|
                |^8$|^9$|^10$|^11$|^12$|^13$|^14$|^15$|</field>
```
As it is necessary to write the ' | ' at the beginning and end of each line that continues the 'or' operation.

Regarding the other issue you are having, it seems like the rule is OK, did you reversed the `Description` field of the rules too? The command `sekurlsa::msv` does create a new process everytime it gets executed?

**Comment by jjf012** [comment]
@cristgl ,According to your suggestion, I modified the local_rule.xml on the manager and re-tested it. The result is as follows:

|command|sysmon event id|wazuh description|
|---|---|---|
| only run `mimikatz.exe` | 1 | 10 ❌ |
| run `lsadump::lsa /inject` in `mimikatz`|10,8,8|only 10 :interrobang: |
| run `sekurlsa::msv` in `mimikatz`| 10 | 10 ⭕️ |

**Comment by cristgl** [comment]
Hello @jjf012,

This is perfectly working at Wazuh 3.9. We have just checked it so that you don't have any problems. The `mimikatz.exe` process generates the Sysmon alert for process creating and, for example, running the command `lsadump::lsa /inject` generates alerts for events 8 and 10 like the following ones:

```
** Alert 1551263981.572643: - sysmon,sysmon_event_10,
2019 Feb 27 11:39:41 (win) any->EventChannel
Rule: 20495 (level 8) -> 'Sysmon - Event 10: ProcessAccess '
{"win":{"system":{"providerName":"Microsoft-Windows-Sysmon","providerGuid":"{5770385F-C22A-43E0-BF4C-06F5698FFBD9}","eventID":"10","version":"3","level":"4","task":"10","opcode":"0","keywords":"0x8000000000000000","systemTime":"2019-02-27T10:39:40.053851200Z","eventRecordID":"173","processID":"3468","threadID":"2032","channel":"Microsoft-Windows-Sysmon/Operational","computer":"pcname","severityValue":"INFORMATION","message":"Process accessed:"},"eventData":{"ruleName":"","utcTime":"2019-02-27 10:39:40.053","sourceProcessGUID":"{A2E669BA-686D-5C76-0000-0010D9CC0E00}","sourceProcessId":"4000","sourceThreadId":"2872","sourceImage":"C:\\Users\\user\\Downloads\\mimikatz_trunk\\x64\\mimikatz.exe","targetProcessGUID":"{A2E669BA-6683-5C76-0000-00104C790000}","targetProcessId":"572","targetImage":"C:\\Windows\\system32\\lsass.exe","grantedAccess":"0x143a","callTrace":"C:\\Windows\\SYSTEM32\\tdll.dll+909ba|C:\\Windows\\system32\\KERNELBASE.dll+2f2e|C:\\Users\\user\\Downloads\\mimikatz_trunk\\x64\\mimikatz.exe+643c9|C:\\Users\\user\\Downloads\\mimikatz_trunk\\x64\\mimikatz.exe+63f53|C:\\Users\\user\\Downloads\\mimikatz_trunk\\x64\\mimikatz.exe+4d28c|C:\\Users\\user\\Downloads\\mimikatz_trunk\\x64\\mimikatz.exe+4d0c4|C:\\Users\\user\\Downloads\\mimikatz_trunk\\x64\\mimikatz.exe+4cea1|C:\\Users\\user\\Downloads\\mimikatz_trunk\\x64\\mimikatz.exe+80675|C:\\Windows\\system32\\KERNEL32.DLL+13d2|C:\\Windows\\SYSTEM32\\tdll.dll+154f4"}}}
win.system.providerName: Microsoft-Windows-Sysmon
win.system.providerGuid: {5770385F-C22A-43E0-BF4C-06F5698FFBD9}
win.system.eventID: 10
win.system.version: 3
win.system.level: 4
win.system.task: 10
win.system.opcode: 0
win.system.keywords: 0x8000000000000000
win.system.systemTime: 2019-02-27T10:39:40.053851200Z
win.system.eventRecordID: 173
win.system.processID: 3468
win.system.threadID: 2032
win.system.channel: Microsoft-Windows-Sysmon/Operational
win.system.computer: pcname
win.system.severityValue: INFORMATION
win.system.message: Process accessed:
win.eventData.utcTime: 2019-02-27 10:39:40.053
win.eventData.sourceProcessGUID: {A2E669BA-686D-5C76-0000-0010D9CC0E00}
win.eventData.sourceProcessId: 4000
win.eventData.sourceThreadId: 2872
win.eventData.sourceImage: C:\Users\user\Downloads\mimikatz_trunk\x64\mimikatz.exe
win.eventData.targetProcessGUID: {A2E669BA-6683-5C76-0000-00104C790000}
win.eventData.targetProcessId: 572
win.eventData.targetImage: C:\Windows\system32\lsass.exe
win.eventData.grantedAccess: 0x143a
win.eventData.callTrace: C:\Windows\SYSTEM32\tdll.dll+909ba|C:\Windows\system32\KERNELBASE.dll+2f2e|C:\Users\user\Downloads\mimikatz_trunk\x64\mimikatz.exe+643c9|C:\Users\user\Downloads\mimikatz_trunk\x64\mimikatz.exe+63f53|C:\Users\user\Downloads\mimikatz_trunk\x64\mimikatz.exe+4d28c|C:\Users\user\Downloads\mimikatz_trunk\x64\mimikatz.exe+4d0c4|C:\Users\user\Downloads\mimikatz_trunk\x64\mimikatz.exe+4cea1|C:\Users\user\Downloads\mimikatz_trunk\x64\mimikatz.exe+80675|C:\Windows\system32\KERNEL32.DLL+13d2|C:\Windows\SYSTEM32\tdll.dll+154f4

** Alert 1551263981.575824: - sysmon,sysmon_event8,
2019 Feb 27 11:39:41 (win) any->EventChannel
Rule: 20493 (level 8) -> 'Sysmon - Event 8: CreateRemoteThread '
{"win":{"system":{"providerName":"Microsoft-Windows-Sysmon","providerGuid":"{5770385F-C22A-43E0-BF4C-06F5698FFBD9}","eventID":"8","version":"2","level":"4","task":"8","opcode":"0","keywords":"0x8000000000000000","systemTime":"2019-02-27T10:39:40.100861100Z","eventRecordID":"174","processID":"3468","threadID":"2032","channel":"Microsoft-Windows-Sysmon/Operational","computer":"pcname","severityValue":"INFORMATION","message":"CreateRemoteThread detected:"},"eventData":{"ruleName":"","utcTime":"2019-02-27 10:39:40.100","sourceProcessGuid":"{A2E669BA-686D-5C76-0000-0010D9CC0E00}","sourceProcessId":"4000","sourceImage":"C:\\Users\\user\\Downloads\\mimikatz_trunk\\x64\\mimikatz.exe","targetProcessGuid":"{A2E669BA-6683-5C76-0000-00104C790000}","targetProcessId":"572","targetImage":"C:\\Windows\\System32\\lsass.exe","newThreadId":"3488","startAddress":"0x0000002704980000","startModule":"","startFunction":""}}}
win.system.providerName: Microsoft-Windows-Sysmon
win.system.providerGuid: {5770385F-C22A-43E0-BF4C-06F5698FFBD9}
win.system.eventID: 8
win.system.version: 2
win.system.level: 4
win.system.task: 8
win.system.opcode: 0
win.system.keywords: 0x8000000000000000
win.system.systemTime: 2019-02-27T10:39:40.100861100Z
win.system.eventRecordID: 174
win.system.processID: 3468
win.system.threadID: 2032
win.system.channel: Microsoft-Windows-Sysmon/Operational
win.system.computer: pcname
win.system.severityValue: INFORMATION
win.system.message: CreateRemoteThread detected:
win.eventData.utcTime: 2019-02-27 10:39:40.100
win.eventData.sourceProcessGuid: {A2E669BA-686D-5C76-0000-0010D9CC0E00}
win.eventData.sourceProcessId: 4000
win.eventData.sourceImage: C:\Users\user\Downloads\mimikatz_trunk\x64\mimikatz.exe
win.eventData.targetProcessGuid: {A2E669BA-6683-5C76-0000-00104C790000}
win.eventData.targetProcessId: 572
win.eventData.targetImage: C:\Windows\System32\lsass.exe
win.eventData.newThreadId: 3488
win.eventData.startAddress: 0x0000002704980000
```

I have the next Sysmon configuration:
```
<?xml version="1.0" encoding="UTF-8"?>
<Sysmon schemaversion="4.10">
   <HashAlgorithms>md5</HashAlgorithms>
   <EventFiltering>
      <!--SYSMON EVENT ID 1 : PROCESS CREATION-->
      <ProcessCreate onmatch="include">
         <Image condition="contains">mimikatz.exe</Image>
      </ProcessCreate>
      <!--SYSMON EVENT ID 2 : FILE CREATION TIME RETROACTIVELY CHANGED IN THE FILESYSTEM-->
      <FileCreateTime onmatch="include" />
      <!--SYSMON EVENT ID 3 : NETWORK CONNECTION INITIATED-->
      <NetworkConnect onmatch="include" />
      <!--SYSMON EVENT ID 4 : RESERVED FOR SYSMON STATUS MESSAGES, THIS LINE IS INCLUDED FOR DOCUMENTATION PURPOSES ONLY-->
      <!--SYSMON EVENT ID 5 : PROCESS ENDED-->
      <ProcessTerminate onmatch="include" />
      <!--SYSMON EVENT ID 6 : DRIVER LOADED INTO KERNEL-->
      <DriverLoad onmatch="include" />
      <!--SYSMON EVENT ID 7 : DLL (IMAGE) LOADED BY PROCESS-->
      <ImageLoad onmatch="include" />
      <!--SYSMON EVENT ID 8 : REMOTE THREAD CREATED-->
      <CreateRemoteThread onmatch="include">
         <SourceImage condition="contains">mimikatz.exe</SourceImage>
      </CreateRemoteThread>
      <!--SYSMON EVENT ID 9 : RAW DISK ACCESS-->
      <RawAccessRead onmatch="include" />
      <!--SYSMON EVENT ID 10 : INTER-PROCESS ACCESS-->
      <ProcessAccess onmatch="include">
         <SourceImage condition="contains">mimikatz.exe</SourceImage>
      </ProcessAccess>
      <!--SYSMON EVENT ID 11 : FILE CREATED-->
      <FileCreate onmatch="include" />
      <!--SYSMON EVENT ID 12 & 13 & 14 : REGISTRY MODIFICATION-->
      <RegistryEvent onmatch="include" />
      <!--SYSMON EVENT ID 15 : ALTERNATE DATA STREAM CREATED-->
      <FileCreateStreamHash onmatch="include" />
      <PipeEvent onmatch="include" />
   </EventFiltering>
</Sysmon>
```

I hope you can try it on v3.9.0 and it helps you.

Kind regards,
Cristina

**Comment by jjf012** [comment]
Then I am waiting for the new version to be released, @cristgl 😄

**Comment by cristgl** [comment]
Hello @jjf012,

When you update to v3.9.0, you should change your rules, we have changed the names as they were too long together, this is the related issue https://github.com/wazuh/wazuh/issues/2680.
With that said, your new rules should look like this:

```
  </rule>
    <rule id="20185" level="8" overwrite="yes">
        <if_sid>20018</if_sid>
        <field name="win.system.eventID">^10$</field>
        <description>Sysmon - Event 10: ProcessAccess $(EventChannel.EventData.Description)</description>
        <options>no_full_log</options>
        <group>sysmon_event_10,</group>
    </rule>
    <rule id="20192" level="8" overwrite="yes">
        <if_sid>20018</if_sid>
        <field name="win.system.eventID">^8$</field>
        <description>Sysmon - Event 8: CreateRemoteThread $(EventChannel.EventData.Description)</description>
        <options>no_full_log</options>
        <group>sysmon_event8,</group>
    </rule>
    <rule id="20194" level="8" overwrite="yes">
        <if_sid>20018</if_sid>
        <field name="win.system.eventID">^1$</field>
        <description>Sysmon - Event 1: Process creation $(EventChannel.EventData.Description)</description>
        <options>no_full_log</options>
        <group>sysmon_event1,</group> 
   </rule>
```
However, they are also included in the new version but with another alert level: [Sysmon rules](https://github.com/wazuh/wazuh-ruleset/blob/2bd86e73bf754337fe2d96d70abe57653f1d8838/rules/0330-sysmon_rules.xml#L122)

**Comment by jjf012** [comment]
@cristgl 
Ok, I will pay attention. It seems that my other fields about Windows rules have to be modified together. 😂

**Comment by Zenidd** [comment]
Hello @jjf012,

Seems like your issue was fixed on the v3.9.0. We hope you were able to adapt your rules satisfactorily.

I'm closing this ticket. If you still have problems do not hesitate to open another one so we can help you.
<br></br>
Best regards,

Juan Pablo Sáez
